rspace/src/main/scala/coop/rchain/rspace/history/ITrieStore.scala

```diff
@@ -18,8 +18,14 @@ trait ITrieStore[T, K, V] {
 
   private[rspace] def getRoot(txn: T, branch: Branch): Option[Blake2b256Hash]
 
+  private[rspace] def persistCheckpointAndGetRoot(txn: T, branch: Branch): Option[Blake2b256Hash]
+
   private[rspace] def putRoot(txn: T, branch: Branch, hash: Blake2b256Hash): Unit
 
+  private[rspace] def validateCheckpointAndPutRoot(txn: T,
```

Comment:
Is this the "safe" version of putRoot? Do we still need the unsafe one?

Reply:
because not all `putRoot` and `getRoot` actions are related to checkpoints. eg. on `delete` the trie did not create a new checkpoint but a new root has been calculated. do you see a way of making this uniform?

Reply:
Yes, agree with @dzajkowski that we definitely still need the "unsafe" one, and I think having two separate functions means it less likely that we would make a mistake when choosing which one to use.